### PR TITLE
PWGHF: new PID selection for candidate-selector-lc-to-k0s-p

### PR DESCRIPTION
**Pull request by DanielSamitz**
- suggestion for additional Bayes PID option in TrackSelectorPID.h
      - uses a cut on the Bayesian probability for selected particle species, below which it is rejected
      - let me know what you think about it
- updated PID selection in candidate-selector-lc-to-k0s-p for bachelor track using the TrackSelectorPID class  
     - related PR https://github.com/AliceO2Group/Run3Analysisvalidation/pull/381

### CODE REVIEW

File: PWGHF/TableProducer/candidateSelectorLcToK0sP.cxx [hunk 1]
```diff
@@ -45,11 +46,16 @@ struct HfCandidateSelectorLcToK0sP {
...
+  Configurable<bool> doBachPid{"doBachPid", true, "Use PID on bachelor track"};
+  Configurable<double> pPidThreshold{"pPidThreshold", 1.0, "Threshold to switch between low and high p TrackSelectors"};
+  Configurable<double> nSigmaTpcMaxLowP{"nSigmaTpcMaxLowP", 2.0, "Max nSigam in TPC for bachelor for low p"};
+  Configurable<double> nSigmaTpcMaxHighP{"nSigmaTpcMaxHighP", 9999., "Max nSigam in TPC for bachelor for high p"};
+  Configurable<double> nSigmaTofMaxLowP{"nSigmaTofMaxLowP", 9999., "Max nSigam in TOF for bachelor for low p"};
+  Configurable<double> nSigmaTofMaxHighP{"nSigmaTofMaxHighP", 3.0, "Max nSigam in TOF for bachelor for high p"};
+  Configurable<bool> requireTofLowP{"requireTofLowP", false, "require TOF information for bachelor for low p"};
+  Configurable<bool> requireTofHighP{"requireTofHighP", true, "require TOF information for bachelor for high p"};
+  Configurable<double> BayesProbLowP{"BayesProbLowP", -1., "threshold for Bayes prob. for bachelor for low p [in %]"};
+  Configurable<double> BayesProbHighP{"BayesProbHighP", -1., "threshold for Bayes prob. for bachelor for high p [in %]"};
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateSelectorLcToK0sP.cxx:58]:
```suggestion
  Configurable<double> probBayesMinLowP{"probBayesMinLowP", -1., "min. Bayes probability for bachelor at low p [%]"};
  Configurable<double> probBayesMinHighP{"probBayesMinHighP", -1., "min. Bayes probability for bachelor at high p [%]"};
```

File: Common/Core/TrackSelectorPID.h [hunk 2]
```diff
@@ -514,6 +514,12 @@ class TrackSelectorPID
     mPtBayesMax = ptMax;
   }
 
+  /// Set cut above which a track should be accepted.
+  void setCutBayes(float cut)
+  {
```
Comment (vkucera) [inline Common/Core/TrackSelectorPID.h:519]:
```suggestion
  void setProbBayesMin(float cut)
  {
```

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by vkucera** [comment]
Thanks @DanielSamitz for this nice extension. Please consider my comments.

**Comment by DanielSamitz** [comment]
Dear @vkucera ,
thank you for your comments.
Both PRs are updated.

Does the clang-format error have something to do with my task? I ran clang-format on all files, and the error I get here says something about install prerequisites?

**Comment by vkucera** [comment]
> 

It's a problem of the GitHub action itself.
@TimoWilken could you have a look why `clang-format` cannot be installed?

**Comment by TimoWilken** [comment]
> @TimoWilken could you have a look why clang-format cannot be installed?

Hi @vkucera @DanielSamitz, I've updated the clang-format workflow, but GitHub isn't picking it up for this PR. You may have to rebase on top of the current `master` branch of O2Physics, sorry!

**Comment by DanielSamitz** [comment]
> Hi @vkucera @DanielSamitz, I've updated the clang-format workflow, but GitHub isn't picking it up for this PR. You may have to rebase on top of the current master branch of O2Physics, sorry!

@TimoWilken OK, worked now. Thank you!

File: Common/Core/TrackSelectorPID.h [hunk 3]
```diff
@@ -532,7 +538,47 @@ class TrackSelectorPID
     return track.bayesID() == mSpecies;
   }
 
-  /// Returns status of Bayesian PID selection for a given track.
+  /// Bayesian probability for particle type.
+  template <typename T>
+  bool isSelectedTrackBayesProbPID(const T& track)
```
Comment (vkucera) [inline Common/Core/TrackSelectorPID.h:543]: The documentation is not complete here.

File: Common/Core/TrackSelectorPID.h [hunk 4]
```diff
@@ -514,6 +514,12 @@ class TrackSelectorPID
     mPtBayesMax = ptMax;
   }
 
+  /// Set cut above which a track should be accepted.
+  void setProbBayesMin(float cut)
```
Comment (vkucera) [inline Common/Core/TrackSelectorPID.h:518]:
```suggestion
  /// Set minimum Bayesian probability above which a track should be accepted.
  void setProbBayesMin(float cut)
```

**Review by vkucera — CHANGES_REQUESTED** [review]
A few details to fix. Otherwise looks good to me.

**Comment by DanielSamitz** [comment]
updated documentation in TrackSelectorPID.h

**Review by vkucera — APPROVED** [review]

**Review by vkucera — APPROVED** [review]

**Comment by vkucera** [comment]
@jgrosseo could you please merge this?

**Review by jgrosseo — APPROVED** [review]

**Pull request merged**